File: -Stack/STACK.java

public class STACK {
    class ListNode {
        int val ;
        ListNode next ;
        ListNode (int val){
            this.val = val;
            this.next = null;
        }
    }
    ListNode head;
    int counter=0;
    void PUSH(int val){
        ListNode temp = new ListNode(val);
        if (head ==null){
            head = temp;
        }
        else {
            ListNode current = head;
            while (current.next!=null){
                current = current.next;
            }
            current.next = temp;
        }
        counter ++;
    }
    ListNode POP(){
        if (head==null ){
            return null;
        }
        ListNode current = head;
        ListNode previous = null;
        while (current.next!=null){
            previous = current ;
            current = current.next;
        }                                 
        ListNode popped = current;
        previous.next = null;
        counter--;
        return popped;
    }
    ListNode PEEK(){
        if (head==null){
            return null;
        }
        ListNode current = head ;
        while (current.next!=null) {
            current = current.next;
        }
        return current;
    }         

    boolean isEMPTY(){
        if (counter == 0) {
            return true;
        }
        else {
            return false;
        }
    }
    void Print(){
        if (head==null){
            return;
        }
        ListNode current = head ;
        System.out.print("[");
        while (current!=null){
            System.out.print(current.val+",");
            current= current.next;
        }
        System.out.println("]");
    }
    public static void main(String[] args) {
        STACK S = new STACK();
        System.out.println("appending values :");
        S.PUSH(12);
        S.PUSH(23);
        S.PUSH(34);
        S.PUSH(45);
        S.Print();
        System.out.println("taking a peek : "+(S.PEEK()).val);
        S.Print();
        System.out.println("popping from stack : "+(S.POP()).val);
        S.Print();
        System.out.println("checking if the stack is empty : "+S.isEMPTY());
        S.Print();
    }
}

File: -Stack/Stack.py
class ListNode :
    data = 0
    next = None
    def __init__(self,data:int) -> None:
        self.data = data
        self.next = None
    head = None
    counter = 0

    def push(self,val:int) -> None :
        temp = ListNode(val)
        if not self.head :
            self.head = temp
        else :
            current = self.head
            while current.next :
                current = current.next
            current.next = temp
        self.counter+=1

    def pop(self):
        if not self.head :
            return
        else :
            previous = None
            current = self.head
            while current.next :
                previous = current
                current = current.next
            popped = current.data
            previous.next = None
        self.counter-=1
        return popped

    def peek(self):
        if not self.head :
            return None
        else :
            current = self.head
            while current.next :
                current = current.next
            return current

    def isEmpty(self) -> bool :
        if self.counter == 0 :
            return True
        else :
            return False

    def Print(self) -> None :
        if not self.head :
            return 
        else:
            current = self.head
            print("[",end=" ")
            while current :
                print(current.data,",",end=" ")
                current = current.next
            print("]")

S = ListNode(None)
